### DISCUSSION TITLE
Name you price

### DISCUSSION THREAD

**Discussion by FutureBuddha** [swellstores/community#528] [Ideas]
Is there already a feature that the user it self can decide on the price of a product?
For example for a donation.
So that the user can decide on the amount he wants to give and maby in case of a subscription can also choose the Intervall?
I mean it is possible if you use the quantity of a donation product but it doesn't really look good on invoices.
If there is already a way implemented please share your way of doing so.

Thank in advance

**Comment by sags95** [comment]
It's not supported through the Frontend API, but it would be possible with the Backend API by updating the cart item with the suggested price: https://developers.swell.is/backend-api/carts/update-a-cart

**Comment by FutureBuddha** [comment]
can you please give me a example i can not really follow your idea.

**Reply by sags95** [reply]
Sure for example, for a name your price feature:

```
swell.init('client-id', 'key');

const updateCart = await swell.put(`/carts/${cart_id}/items/${id}`, {
    price: myPrice
});
```

**Reply by FutureBuddha** [reply]
wonder full and this would be than Independent from the price i set for the items or products or how would it be handelt in the backend by checkout or after checkout in the order?

**Reply by sags95** [reply]
Correct, it will still pass through checkout without issue as it's essentially overriding the `price` field. 

To add it to the cart with custom price (instead of update) it would be:
```
swell.init('client-id', 'key');

const addToCart = await swell.put(`/carts/${cart_id}/items`, {
    product_id: $product,
    price: $price,
    ...
});
```

**Comment by FutureBuddha** [comment]
and would the same work with a subscription where i can manually set the amount and interval of the subscription?

**Reply by sags95** [reply]
Similar concept, but instead you would be passing in the the `purchase_option` fields for price and interval.
